pagefaultgames/pokerogue
# [UI][Dev] Improve event banner placement

## What are the changes the user will see?
- Event banner and timer gets centered automatically based on the menu size in the used language
- In the event the main menu takes a lot of space like in Italian, a small portion of the left and right of the banner will be hidden instead of a big part of the right side being hidden
- (dev side) no more fiddling with numbers to place the banner when setting up a new event


## Why am I making these changes?
The person adding an event shouldn't have to deal with magic numbers like these `this.banner.setOrigin(0.08, -0.35);` 





## What are the changes from a developer perspective?
This is the quick & dirty version of what i made initially, so. yeah.
- Renamed "generic" event type to clarify what it actually does
- Title screen provides the TimedEvenContainer with the space it can use
- Event banner and timer gets centered in that space, starting from the bottom
- Event banner gets offset to make room for event timer if it's to be shown

(My initial solution was to make the banner alignment (left, center, right) a parameter and a few other things, but that code's gone so for now this will have to do)




### Screenshots/Videos 
<details><summary>couple screenshots</summary>

![Screenshot_20241026-150008](https://github.com/user-attachments/assets/d80ed986-a79d-422a-8162-cf392de32065)
![Screenshot_20241026-152359](https://github.com/user-attachments/assets/7fe3316b-9e70-4812-9a98-a149fcf3ec5d)
This looks off with the big menu because those screenshots use an old banner that has its text on the left.
![Screenshot_20241026-150137](https://github.com/user-attachments/assets/6ca6d27b-f474-44ca-98a9-acb7a90cf443)

</details>

For before/after screenshots with a banner that actually needs to be centered: [see here on discord](https://discord.com/channels/1125469663833370665/1295531626465595402/1298366551942103072) (need contributor rights)




## How to test the changes?
in `timed-event-manager' lines 29-32:
- change the endDate so that the event is considered ongoing
- change the EventType to SHINY to have the timer display, or leave it as is so it doesn't

Start the game, switch languages, make sure the event banner and timers are shown properly







## Checklist
